crates/api/src/local_user/save_settings.rs

```diff
@@ -119,21 +118,20 @@ impl Perform for SaveUserSettings {
     })
     .await?
     .map_err(|e| LemmyError::from_error_message(e, "user_already_exists"))?;
-    let discussion_languages: Option<Vec<LanguageId>> =
-      if let Some(discussion_languages) = data.discussion_languages.clone() {
-        let mut language_ids = vec![];
-        for l in discussion_languages {
-          language_ids.push(
-            blocking(context.pool(), move |conn| {
-              Language::read_id_from_code(conn, l)
-            })
-            .await??,
-          );
-        }
-        Some(language_ids)
-      } else {
-        None
-      };
+
+    if let Some(discussion_languages) = data.discussion_languages.clone() {
+      for language_id in discussion_languages {
+        let form = LocalUserLanguageForm {
+          local_user_id,
+          language_id,
+        };
+
+        blocking(context.pool(), move |conn| {
+          LocalUserLanguage::create(conn, &form)
+        })
+        .await??;
+      }
+    }
```

Comment:
I dont think this is correct, because it doesnt remove languages which the user unselected.

Reply:
Ah good call.